Automattic/wp-calypso
# A4A: Fix broken A4A site action links.

This pull request addresses broken links in the Site dashboard for Simple sites, Jetpack, and A4A client sites. Currently, the Site actions component from Jetpack Manage is being reused, but these links are not applicable to the A4A dashboard.

<img width="393" alt="Screenshot 2024-07-23 at 3 16 54 PM" src="https://github.com/user-attachments/assets/94c61c22-4463-4440-b13c-3e96bf0410bb">


Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/686

## Proposed Changes

* This PR creates A4A-specific site action components for the Site dashboard. We created a separate one to avoid bloating the Jetpack Manage's site action component, which leads to more complicated and unmanageable code.
* **Additionally, the PR removes the remnant of A4A-specific conditional flow in the Jetpack Manage's site actions component.**

## Why are these changes being made?

* Broken links does not look good in our Dashboard.

## Testing Instructions

* Use the A4A live link below and go to the `/sites` page.
* Please ensure that WPCOM atomic sites, Jetpack, A4A sites, and WPCOM simple sites are all connected to your dashboard. If you're wondering how to add a Simple site to the dashboard, you will need to revoke the license of a WPCOM atomic site provisioned in A4A. Revoking the license will downgrade the site to a Simple site.
* Test that the Site action links are working and has no broken links for all types of sites.
* **Finally, test Jetpack Manage's site action for regression.**


## Pre-merge Checklist
